src/components/tables/Table.tsx

```diff
@@ -22,6 +22,7 @@ import { ITableCompositeState, ITableData } from './TableReducers';
 
 export interface IData {
   id: string;
+
   [attribute: string]: any;
 }
```

Comment:
I wonder if it would be nice to rewrite this as follows:

```
 export type IData<T> = T & {
   id: string;
 }
```  
It would definitely help with the types when using the table. See https://www.typescriptlang.org/docs/handbook/advanced-types.html for more info. It wouldn't be backward compatible though so not sure it's the best idea :(